src/App.vue

```diff
@@ -593,26 +593,28 @@ export default {
         .then(response => Promise.resolve(response.data))
     },
 
-    storeNrData (nr: any): void {
-      // check if NR is valid
-      if (!this.isNrValid(nr)) {
-        this.nameRequestInvalidDialog = true
-        throw new Error('Invalid NR data')
-      }
+    storeNrData (nr: any, ia: any): void {
+      if (ia.filing.header.status in [FilingStatus.DRAFT, FilingStatus.PENDING]) {
```

Comment:
I'm trying to think what states the filing could be in, and what should be done in various cases...
Here, you've determined that we should check the NR only if the filing is Draft or Pending. What about the other states?
A couple of options to make this code "fuller": add a comment explaining the business rule here; or use a switch-case (with comments) to itemize what happens for all states. Or ...?

Reply:
Or change this condition to "don't check the NR if the filing is Completed"?

What do you think?

Reply:
I think it should be fine to check !Completed. Will change it